### DISCUSSION TITLE
How to customise maxAge for state and pkce cookies?

### DISCUSSION THREAD

**Discussion by jpogorzelski** [nextauthjs/next-auth#8233] [Help]
Currently, the `maxAge` property of `state` cookie is set to 900 = 15 minutes. The same goes for `pkce` cookie. I'd like to customize this value because of the error handling - the NextAuth throws the error
```
[next-auth][error][OAUTH_CALLBACK_ERROR] 
https://next-auth.js.org/errors#oauth_callback_error State cookie was missing.
```
even after successful authorization from OAuth2 provider. NextAuth should not fail at this point, as it is bad user experience.

I am familiar with the issue https://github.com/nextauthjs/next-auth/issues/4660 and the PR https://github.com/nextauthjs/next-auth/pull/4719 but there is no information how to configure it properly using NextAuth options.
I tried following:
```
cookies: {
  state: {
    options: {
      maxAge: 1800
    }
  }
}
```

and indeed it sets the cookie with this age, but the `name` is undefined as well as other options.
I'd like to change only this property and leave the rest as default, especially I'd prefer not to tinker with the cookie name as it's name also depends on the `secure` option: `${cookiePrefix}next-auth.state`.

Ideally, I think we could use an environment variable for this.

**Comment by sawyerh** [comment]
@jpogorzelski Do you remember how to reproduce the `OAUTH_CALLBACK_ERROR` due to the state cookie's expiration? Trying to reproduce this same error and am trying to understand if its always due to what you're mentioning here.

**Reply by jpogorzelski** [reply]
1. First, look at the state cookie, `__Secure-next-auth.state`. It has max age of 15 min, e.g. `2024-01-09T12:39:01.010Z`.
2. Second, look at the max time the OAuth2 provider would wait for the response. In my case it's Ory Hydra, which has cookie with max age of 30 min (e.g `2024-01-09T12:54:01.042Z`).

To reproduce the issue like normal user would, just stay at login page (for whatever OAuth2 provider you are using) longer than cookie expiry time from no. 1 (15 min in my case), but shorter that expiry time from no. 2 (30 min for me).

Or you can force the issue by removing / reducing max age value of `__Secure-next-auth.state` cookie.
